sql-odbc/src/sqlodbc/opensearch_connection.cpp

```diff
@@ -95,6 +95,15 @@ int LIBOPENSEARCH_connect(ConnectionClass *self) {
     rt_opts.crypt.verify_server = (self->connInfo.verify_server == 1);
     rt_opts.crypt.use_ssl = (self->connInfo.use_ssl == 1);
 
+    // Server host url protocol is necessary for sql plugin validation
+    if (rt_opts.conn.server.size() && rt_opts.conn.server.find("http", 0) != 0) {
```

Comment:
This `if` statement looks unnecessary -- `rt_opts` is initialized from default values here.

Reply:
Not sure what you mean. The purpose of this `if` statement is to determine if there is no protocol specified in the url. We have to handle both cases for the url protocol. If no protocol is prepended then the SQL plugin query will fail with an SSL enabled node.

Reply:
I see. I didn't notice `server` being set on line 84.

My high-level request is to set `rt_opts.conn.server` once in this function. The logic for determining what to set `rt_opts.conn.server` from `connInfo` is complex enough to be in its own function, I think.

What should happen if `useSSL = 1` and `connInfo.server` is `http://example.com`?

Reply:
Windows users will have a UI block that does not allow the `UseSSL` flag to differ from the url specified protocol. Mac users are able to create this edge case due to no UI fix. I tried to make a generic rule to make it impossible to use differing protocols, but this broke my UI fix for windows users. 
[864ee58a6e76d70e30dabb0aac1f7aae2f14048c](https://github.com/Bit-Quill/opensearch-project-sql/pull/75/commits/864ee58a6e76d70e30dabb0aac1f7aae2f14048c)